src/Apps/Order/Routes/__tests__/Reject.test.tsx

```diff
@@ -82,4 +94,82 @@ describe("Buyer rejects seller offer", () => {
       `/orders/${testOrder.id}/respond`
     )
   })
+
+  describe("mutation", () => {
+    beforeEach(() => {
+      commitMutation.mockReset()
+    })
+
+    it("routes to status page after mutation completes", () => {
+      const component = getWrapper()
+      const mockCommitMutation = commitMutation as jest.Mock<any>
+      mockCommitMutation.mockImplementationOnce(
+        (_environment, { onCompleted }) => {
+          onCompleted(rejectOfferSuccess)
+        }
+      )
+      const submitButton = component.find(Button).last()
+      submitButton.simulate("click")
+
+      expect(mockPushRoute).toHaveBeenCalledWith(
+        `/orders/${testOrder.id}/status`
+      )
+    })
+
+    it("shows the button spinner while loading the mutation", () => {
+      const component = getWrapper()
+      const mockCommitMutation = commitMutation as jest.Mock<any>
+      mockCommitMutation.mockImplementationOnce(() => {
+        const buttonProps = component
+          .update()
+          .find("Button")
+          .props() as any
+        expect(buttonProps.loading).toBeTruthy()
+      })
+
+      const submitButton = component.find(Button).last()
+      submitButton.simulate("click")
+    })
+
+    it("hides the button spinner when the mutation completes", () => {
+      const component = getWrapper()
+      const mockCommitMutation = commitMutation as jest.Mock<any>
+      mockCommitMutation.mockImplementationOnce(
+        (_environment, { onCompleted }) => {
+          onCompleted(rejectOfferSuccess)
+        }
+      )
+      const submitButton = component.find(Button).last()
+      submitButton.simulate("click")
+
+      const buttonProps = component
+        .update()
+        .find("Button")
+        .props() as any
+      expect(buttonProps.loading).toBeFalsy()
+    })
+  })
+
+  it("shows an error modal when there is an error from the server", () => {
```

Comment:
I think I missed this in other PR reviews, but let's add another test case for something like a network error that would go through the `onError` handler. See [here](https://github.com/artsy/reaction/blob/40f0fc58ee373765ae1248906e4abf63ef0458b5/src/Apps/Order/Routes/__tests__/Review.test.tsx#L112-L121) for an example.

Reply:
Fixed. We can share this test case with _accept_ and _counter_ when we consolidate the code for thos features.